src/worker.c

```diff
@@ -103,6 +123,31 @@ static int init_worker(struct worker_data *data)
 	}
 
 
+        /* Enable periodic worker reporting if enabled on CLI */
+        if (data->worker_report_interval > 0) {
+                fprintf(stderr, "Enable reporting on write pipe fd: %d\n", data->pipe_w); 
+                if ((res = curl_easy_setopt(data->curl, CURLOPT_PROGRESSFUNCTION, old_prog_report)) != CURLE_OK) {
+                        fprintf(stderr, "cURL option error: %s\n", curl_easy_strerror(res));
+                } 
+                if ((res = curl_easy_setopt(data->curl, CURLOPT_PROGRESSDATA, data)) != CURLE_OK) {
+                        fprintf(stderr, "cURL option error: %s\n", curl_easy_strerror(res));
+                } 
+
+#if LIBCURL_VERSION_NUM >= 0x72000
+                if ((res = curl_easy_setopt(data->curl, CURLOPT_XFERINFOFUNCTION, prog_report)) != CURLE_OK) {
+                        fprintf(stderr, "cURL option error: %s\n", curl_easy_strerror(res));
+                } 
+                if ((res = curl_easy_setopt(data->curl, CURLOPT_XFERINFODATA, data)) != CURLE_OK) {
+                        fprintf(stderr, "cURL option error: %s\n", curl_easy_strerror(res));
+                } 
+#endif
```

Comment:
Right now you're adding both report functions for new versions of curl. The old ones should probably be in an #else, no?

The XFERINFOFUNCTION option was added in curl 7.32 which was released more than 4 years ago, though, so feel free to just skip the compatibility stuff and only support that.